numbat-cli/src/main.rs

```diff
@@ -24,12 +24,41 @@ use rustyline::{
     Validator,
 };
 use rustyline::{EventHandler, Highlighter, KeyCode, KeyEvent, Modifiers};
+use serde::Deserialize;
+use toml;
 
 use std::io::IsTerminal;
 use std::path::PathBuf;
 use std::sync::{Arc, Mutex};
 use std::{fs, thread};
 
+#[derive(Deserialize, PartialEq, Eq)]
+#[serde(rename_all = "kebab-case")]
+enum ExchangeRateFetchingPolicy {
+    Prefetch,
+    FetchOnFirstUse,
+    DontFetch,
+}
+
+impl Default for ExchangeRateFetchingPolicy {
+    fn default() -> ExchangeRateFetchingPolicy {
+        ExchangeRateFetchingPolicy::Prefetch
+    }
+}
+
+#[derive(Deserialize, Default)]
+#[serde(rename_all = "kebab-case")]
+struct Config {
+    exchange_rate_fetching_policy: ExchangeRateFetchingPolicy,
+    quiet: bool,
+    #[serde(default = "default_prompt")]
+    prompt: String,
```

Comment:
Just noticed that with this, the derived `default` method returns a config with an empty prompt; not sure how to fix it without restoring to manually writing a `Default` impl...

Reply:
I think the `default_prompt` option is okay. Another thing you can do is to add a `struct Prompt(String)` and implement `Default` for that, but I think that's overkill here.